protocol/verification.go

```diff
@@ -0,0 +1,51 @@
+package protocol
+
+import (
+	"bytes"
+
+	"github.com/coniks-sys/coniks-go/crypto/sign"
+	"github.com/coniks-sys/coniks-go/crypto/vrf"
+	m "github.com/coniks-sys/coniks-go/merkletree"
+)
+
```

Comment:
I thought that the client should do it itself, but you're right. It makes more sense to include this check in the protocol.